plugins/defaultplugins/l3plugin/route_config.go

```diff
@@ -97,15 +115,50 @@ func (plugin *RouteConfigurator) ConfigureRoute(config *l3.StaticRoutes_Route, v
 // ModifyRoute processes the NB config and propagates it to bin api calls.
 func (plugin *RouteConfigurator) ModifyRoute(newConfig *l3.StaticRoutes_Route, oldConfig *l3.StaticRoutes_Route, vrfFromKey string) error {
 	plugin.Log.Infof("Modifying route %v -> %v", oldConfig.DstIpAddr, oldConfig.NextHopAddr)
-	// Validate old route data Vrf.
-	if err := plugin.validateVrfFromKey(oldConfig, vrfFromKey); err != nil {
+
+	outgoingIfName := newConfig.OutgoingInterface
+	if outgoingIfName != "" {
+		_, _, existsNewOutgoing := plugin.SwIfIndexes.LookupIdx(outgoingIfName)
+		routeId := routeIdentifier(oldConfig.VrfId, oldConfig.DstIpAddr, oldConfig.NextHopAddr)
+		if existsNewOutgoing {
+			plugin.Log.Debugf("Route %s unregistered from cache.", routeId)
+			plugin.RouteCachedIndex.UnregisterName(routeId)
+		} else {
+			routeIdx,_, isRouteCached := plugin.RouteCachedIndex.LookupIdx(routeId)
+			if isRouteCached {
+				plugin.RouteCachedIndex.RegisterName(routeId, routeIdx, newConfig)
```

Comment:
missing plugin.RouteIndexSeq++

Reply:
reason is that this is branch where route was found in cache. It means that it exists and that it already has assigned sequence number. It is in case where priority of weight (non key values) has been changed.